tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -21,6 +23,24 @@ using namespace shogun;
 #include <shogun/mathematics/eigen3.h>
 using namespace Eigen;
 
+//function to create sparse matrix
+template<typename T>
+SGSparseMatrix<T> GenerateSparseMatrix(const int32_t numberOfFeatures,
+                                       const int32_t numberOfVectors,
+                                       const float64_t sparseLevel)
```

Comment:
Why making all these const? They are passed by value anyway. It's fine, just wondering.